spring-kafka/src/test/java/org/springframework/kafka/core/DefaultShareConsumerFactoryTests.java

```diff
@@ -197,4 +185,112 @@ void integrationTestDefaultShareConsumerFactory(EmbeddedKafkaBroker broker) thro
 		consumer.close();
 	}
 
+	@Test
+	void integrationTestSharedConsumersDistribution(EmbeddedKafkaBroker broker) throws Exception {
+		final String topic = "embedded-share-distribution-test";
+		final String groupId = "distributionTestGroup";
+		int recordCount = 8;
+		List<String> consumerIds = List.of("client-dist-1", "client-dist-2");
+		Map<String, Set<String>> consumerRecords = runSharedConsumerTest(topic, groupId, consumerIds,
+				recordCount, broker);
+
+		// Assert all records were received (no loss)
+		Set<String> allReceived = new java.util.HashSet<>();
+		consumerRecords.values().forEach(allReceived::addAll);
```

Comment:
This still ha not been fixed.
When we put in set a value which has been already there, nothing happens with the set.
Therefore an assertion against set does not reflect a reality and does not prove that parallel consumers receive same data. Or did I misunderstood what is going on here?